SmartPlaces.Facilities/samples/Telemetry.Mapped/src/Processors/TelemetryIngestionProcessor.cs

```diff
@@ -318,47 +323,65 @@ private dynamic TranslateType(TypedValue sourceData, DTEntityKind targetType)
                         case DTEntityKind.String:
                             return sourceData.EnumValue.ToString();
                     }
+
                     break;
                 case TypedValue.ValueOneofCase.Float32Value:
                     switch (targetType)
                     {
                         case DTEntityKind.Double:
-                            return Convert.ToDouble(sourceData.Float32Value);
+                            if (sourceData.Float32Value.ToString() != "NaN")
```

Comment:
Are there any other double strings that we might have issues with? `+INF` or `-INF`? I've seen `NaN` before, but not in dotnet, so I'm a little fuzzy what is even possible to fill this space.

Reply:
Potentially we could use:
```csharp
var convertDouble = Convert.ToDouble(sourceData.Float32Value);
if(Double.IsFinite(convertDouble))
{
    return convertDouble;
}
else 
{
    throw new InvalidCastException($"Failed to convert '{convertDouble}' to {targetType}");
}
```
https://dotnetfiddle.net/9DlvaH